libs/langchain-anthropic/src/experimental/utils/tool_calling.ts

```diff
@@ -100,7 +100,10 @@ export function fixArrayXMLParameters(
       typeof xmlParameters[key] === "object" &&
       xmlParameters[key] !== null
     ) {
-      fixedParameters[key] = fixArrayXMLParameters(schema, xmlParameters[key]);
+      fixedParameters[key] = fixArrayXMLParameters(
+        schema.properties[key] as JsonSchema7ObjectType,
```

Comment:
@davidfant had:

```
{
        ...schema.properties[key],
        definitions: schema.definitions,
}
```

for the first arg - does that make sense here?

Reply:
This will break JSON schema definitions. TBH I'm not sure how well this XML situation would even work with JSON schema definitions though, so it might not practically matter, but I'd still prefer adding in definitions

Reply:
Will change it to yours